### ISSUE TITLE
Stop using MD5?

### ISSUE THREAD

**Issue by commutateigh** [jjjake/internetarchive#315]
It looks like IA items provide CRC32, MD5, and SHA-1 hashes, which are all insecure hash functions. I suggest that we use the least insecure (SHA-1) instead of MD5 for checking integrity. If it's there, you might as well use the best one, right?

Since all those functions aren't great, another option might be to use all of them at once, and maybe the size of the file as well. That would probably be the most robust unless the IA starts using non-archaic hash functions, [which is unlikely](https://help.archive.org/hc/en-us/articles/360016553532-Live-Music-Archive-etree-org-#checksum-file): 

> Systems have improved to the point where we no longer see checksums as necessary to verify accurate upload/download.

TL;DR: Abandon MD5, adopt SHA-1.

**Comment by JustAnotherArchivist** [comment]
While collision resistance of MD5 and SHA-1 is broken, it doesn't matter here. The hashes are only to check against corruption during transfer or storage. Random corruption due to equipment or software errors are extremely unlikely to not alter the hash, to the point that it's not even worth considering. Malicious modification while keeping the hash fixed is infeasible as well; nobody has broken the hash functions' preimage resistance yet. Even MD4 and MD2 aren't considered broken in that respect.

You're many orders of magnitude more likely to calculate a hash for corrupt data due to a hard drive or memory bit flip than to have a matching hash but the wrong data.

**Comment by jjjake** [comment]
I agree with what @JustAnotherArchivist said, and this would need to be changed at the IA-S3 API level. Thanks for the feedback though, @commutateigh!
